homeassistant/components/nobo_hub/climate.py

```diff
@@ -0,0 +1,407 @@
+        """Set HVAC mode to comfort(HEAT) or back to normal(AUTO)."""
+        if hvac_mode == HVAC_MODE_AUTO:
+            await self.async_set_preset_mode(PRESET_NONE)
+            self._current_mode = hvac_mode
+        elif hvac_mode == HVAC_MODE_HEAT:
+            await self.async_set_preset_mode(PRESET_COMFORT)
+            self._current_mode = hvac_mode
+
+        if self.can_turn_off():
+            if hvac_mode == HVAC_MODE_OFF:
+                await self.async_set_preset_mode(PRESET_NONE)
+                self._current_mode = hvac_mode
+                await self._nobo.async_update_zone(
+                    self._id, week_profile_id=self._command_off_id
+                )  # Change week profile to OFF
+                _LOGGER.debug(
+                    "Turned off heater %s '%s' by switching to week profile %s",
+                    self._id,
+                    self._name,
+                    self._command_off_id,
+                )
+            else:
+                await self._nobo.async_update_zone(
+                    self._id, week_profile_id=self._command_on_id
+                )  # Change week profile to normal for this zone
+                _LOGGER.debug(
+                    "Turned on heater %s '%s' by switching to week profile %s",
+                    self._id,
+                    self._name,
+                    self._command_on_id,
+                )
+            # When switching between AUTO and OFF an immediate update does not work (the Nobø API seems to answer with old values), but it works if we add a short delay:
+            await asyncio.sleep(0.5)
+        elif hvac_mode == HVAC_MODE_OFF:
+            _LOGGER.warning(
+                "User tried to turn off zone %s '%s', but this is not configured so this should be impossible",
+                self._id,
+                self._name,
+            )
+
+    def can_turn_off(self) -> bool:
+        """Return true if heater can turn off and on."""
+        return self._command_on_id is not None and self._command_off_id is not None
+
+    async def async_set_preset_mode(self, preset_mode):
+        """Set new zone override."""
+        if self._nobo.zones[self._id][ATTR_OVERRIDE_ALLOWED] == "1":
+            if preset_mode == PRESET_ECO:
+                mode = self._nobo.API.OVERRIDE_MODE_ECO
+            elif preset_mode == PRESET_AWAY:
+                mode = self._nobo.API.OVERRIDE_MODE_AWAY
+            elif preset_mode == PRESET_COMFORT:
+                mode = self._nobo.API.OVERRIDE_MODE_COMFORT
+            else:  # PRESET_NONE
+                mode = self._nobo.API.OVERRIDE_MODE_NORMAL
+            await self._nobo.async_create_override(
+                mode,
+                self._nobo.API.OVERRIDE_TYPE_CONSTANT,
+                self._nobo.API.OVERRIDE_TARGET_ZONE,
+                self._id,
+            )
+            # TODO: override to program if new operation mode == current week profile status
```

Comment:
@echoromeo How did you intend to do this?

Reply:
I figured out a way to do it, but when looking at the climate UI, I think it makes sense to keep the current behaviour and just remove the todo.